modules/menu/dynamic_menu.c

```diff
@@ -228,6 +228,54 @@ static int call_xfer(struct re_printf *pf, void *arg)
 }
 
 
+static int call_att_xfer(struct re_printf *pf, void *arg)
+{
+	struct menu *menu = menu_get();
+	struct call *call = uag_call_find(menu->callid);
+
+	if (menu->attended_callid) {
+		re_hprintf(pf, "attended transfer already in process\n");
+		return ENOENT;
+	}
+	else {
+		if (!call_is_onhold(call)) {
+			re_hprintf(pf, "please hold your active call\n");
+			return ENOENT;
+		}
+		str_dup(&menu->attended_callid, menu->callid);
+		dial_handler(pf, arg);
```

Comment:
The function `dial_handler()` is very high-level. And it should not be used from `dynamic_menu.c`.

The command `att_transfer` resp. function `call_att_xfer` should be put to static_menu.c. But is it really needed? Further, do we really need variable `attended_callid`? It would be also possible that the user specifies the `callid` for the `attended_call` in function `transfer_att_xfer()`.

So the attended transfer could start by calling command `dial`. Then baresip prints/returns the call-id (see UA events!).
Then user calls `forw_transfer` with the call-id.

Reply:
So change the workflow to
- put active call on hold
- start a new call via dial
- transfer the calls with a call-id parameter

I wanted to keep it simple for the user but I am open for changes.

Reply:
We have two options for the second call.
- Full flexibility by using command `dial` or `dialdir` from static_menu.c
   - This approach needs less changes.
   - All features for an outgoing call like selecting audio/video dir, SIP auto answer. But maybe this is not needed for an attended transfer and only confusing.
- Your `call_att_xfer` in dynamic_menu.c would allow more control for this use-case.
   - I agree that this should stay in the dynamic_menu.c, because another call (the first call) has to be present already.
   - This function should not use `dial_handler`.
   - Select of the current call is ok. But an optional argument `[callid=id]` would be advantageous.
   - The ua is then `call_ua(call)`.
   - Use `account_uri_complete()` and `ua_connect()` instead of `dial_handler()`!
   - Maybe we should also hold the call if it is not on-hold.

Note: `re_hprintf` is ok.

Reply:
On my rewrite I will try the first option.

If you think that the callid could change unexpectedly should the ids of both calls be parameter of the `att_xfer` function?

Reply:
Ok.

Yes, for the first option the variable `attended_callid` should be removed and given as parameter. The second callid could be optional. Add also a usage text like in `cmd_answerdir()`! It should be enough to support this variant:
```
/atransfer <from-id> [to-id]
```
which calls your `transfer_att_xfer()`.

Reply:
if I put one call on hold and directly call the transfer it seems that _sometimes_ there are problems with the transfer.

I cannot reproduce it constantly, so I leave the holding to the user.